openbas-api/src/main/java/io/openbas/executors/crowdstrike/client/CrowdStrikeExecutorClient.java

```diff
@@ -51,19 +51,76 @@ public class CrowdStrikeExecutorClient {
 
   // -- ENDPOINTS --
 
-  public ResourcesHosts devices() {
+  public List<CrowdStrikeDevice> devices(String hostGroup) {
+    try {
+      int offset = 0;
+      List<CrowdStrikeDevice> hosts = new ArrayList<>();
+      ResourcesHosts partialResults = getResourcesHosts(offset, hostGroup);
+      if (partialResults.getResources() == null) {
+        return hosts;
+      } else {
+        hosts.addAll(partialResults.getResources());
+      }
+      while (partialResults.getMeta().getPagination().getTotal() > offset) {
```

Comment:
I'm always afraid of end condition with while loop for some reasons so, what do you think of this (untested) :
```suggestion
      int numberOfExecution = Math.ceilDiv(partialResults.getMeta().getPagination().getTotal(), partialResults.getMeta().getPagination().getLimit());
      for (int callNumber = 1; callNumber < numberOfExecution; callNumber += 1) {
```